core/src/main/java/org/infinispan/util/CacheStreamMapper.java

```diff
@@ -0,0 +1,25 @@
+package org.infinispan.util;
+
+import java.util.Set;
+
+import org.infinispan.CacheStream;
+import org.infinispan.commons.util.InjectiveFunction;
+
+/**
+ * CacheStream that allows for mapping the filtered keys to different keys. Note that the function provided
+ * must be a {@link InjectiveFunction} guaranteeing that keys are distinct when mapped.
+ * @author wburns
+ * @since 9.2
+ */
+class CacheStreamMapper<R> extends AbstractDelegatingCacheStream<R> {
```

Comment:
I'd rename it to `MapperCacheStream`. And maybe check that `keyMapper instanceof InjectiveFunction`?

Reply:
Sure about name. Not sure what you mean about the second point, the `keyMapper` has to be an instanceof `InjectiveFunction` as it is declared in the constructor.

Reply:
I must admit `MapperCacheStream` sounds a bit odd to me. Maybe `MappedCacheStream` ?